Python/import.c

```diff
@@ -396,15 +396,18 @@ static const char * const sys_files[] = {
 void
 PyImport_Cleanup(void)
 {
+    assert(!PyErr_Occurred());
+
     Py_ssize_t pos;
     PyObject *key, *value, *dict;
     PyInterpreterState *interp = PyThreadState_GET()->interp;
-    PyObject *modules = PyImport_GetModuleDict();
     PyObject *weaklist = NULL;
     const char * const *p;
 
-    if (modules == NULL)
+    PyObject *modules = interp->modules;
```

Comment:
I'd rather avoid this, if possible.  Under what conditions does the use of PyImport_GetModuleDict() cause a fatal error?

Reply:
If you call PyImport_Cleanup() twice. In Python 3.6, calling PyImport_Cleanup() a second time does nothing: that's the purpose of the test on the next line.

PyImport_Cleanup() is part of the public API, even if it's documented as "please don't call this function" ;-) I would prefer to restore Python 3.6 behaviour.

Reply:
Ah, thanks for clarifying.  LGTM then.  I still plan on landing #3606 at some point, but at that point I'll have to deal with recent ``interp->modules`` usage anyway. :)